torch/_functorch/_aot_autograd/subclass_utils.py

```diff
@@ -54,15 +55,19 @@ def maybe_suggest_memory_format(
     return suggest_memory_format(t)
 
 
-def get_types_for_subclass(tensor_subclass):
-    if not is_traceable_wrapper_subclass(tensor_subclass):
-        return ["Tensor"]
-    inner_keys, _ = tensor_subclass.__tensor_flatten__()
-    result = []
-    for key in inner_keys:
-        inner_tensor = getattr(tensor_subclass, key)
-        result.extend(get_types_for_subclass(inner_tensor))
-    return result
+def get_types_for_subclass(tensor_subclass) -> Dict[Any, List[Any]]:
```

Comment:
can we do better than Any -> Any? It seems like this is actually `Dict[type, List[torch.Tensor]]`?

From the function name it's not immediately clear why we're returning a dict - maybe worth a quick comment on the top saying that the values are all of the instances that we found of each subclass type